rrmngmnt/firewall.py

```diff
@@ -144,6 +144,25 @@ def add_rule(self, dest, target, protocol='all', ports=None):
             protocol, ports
         )
 
+    def insert_rule(self, dest, target, protocol='all', ports=None):
+        """
+        Insert new firewall rule to a specific chain
+
+        Args:
+            dest (dict): 'address' key and value containing destination host or
+               list of destination hosts
+            target (str): Target rule to apply
+            protocol (str): affected network protocol, Default is 'all'
+            ports (list): list of ports to configure
+
+        Returns:
+            bool: False if inserting new rule failed, True if it succeeded
+        """
+        return self.edit_chain(
```

Comment:
Maybe better to set by rule number.
iptables -D FORWARD 1, for example, this one will be the first one.

Reply:
I do not sure, but I think it works only for delete.

Reply:
iptables -I INPUT 1 -i lo -j ACCEPT
https://serverfault.com/questions/374993/how-do-you-prepend-an-iptables-rather-than-append

Reply:
I've added rule number. Please have a look.